tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -133,12 +133,20 @@ internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLeve
                 parameter.DbType = DbType.Binary;
                 injectionCommand.Parameters.Add(parameter);
 
-                injectionCommand.ExecuteNonQuery();
-
                 if (Log.IsEnabled(LogEventLevel.Debug))
                 {
                     // avoid building the string representation in the general case where debug is disabled
-                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                    Log.Debug("Propagating span data for DBM for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                }
+
+                try
+                {
+                    injectionCommand.ExecuteNonQuery();
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Error setting context_info [{ContextValue}] for DB query, falling back to service only propagation mode. There won't be any link with APM traces.", HexConverter.ToString(contextValue));
+                    return false;
                 }
```

Comment:
So technically we'd be in `service` mode because the initial `Comment` injection worked prior to this, but I'm wondering if we want to opt out of attempting `PropagateDataViaContext` again on the next go. As it is currently I think we'd just constantly attempt this - maybe that make sense as we've seen it succeed in more cases then it has failed?
But I'm worried for the cases where it is failing constantly.

Reply:
@andrewlock do you know if there is rate limiting for the `RedactedErrorLogCollector`? From looking at it, maybe, but not really? As there is some upper limits set on the queue

Reply:
I asked on slack, I added a limit after a fixed number of logs